OwO mod overhaul!

Teo45dore opened the pull request:
Major Changes/Fixes:
- Merged IL2CPP/Mono projects into a single cross-compatible project.
- Changed all OwO classes to use harmony for IL2CPP/Mono cross-compatibility.
- Fixed a TypeLoadException that would frequently occur in old IL2CPP version.
- Removed the ```PatchUtils``` class because after switching to just using Hamony it became redundant.
- Because of the switch away from native hooking and IL2CPP shenanigans the mod no longer depends on UnhollowerBaseLib.

Minor Changes/Fixes:
- Started caching already OwOed strings in a ```Dictionary<string, string>```  to prevent flickering of owostrings and owofaces for text components that are frequently updated.
- Switched mod initialization from the depreciated ```MelonMod.OnApplicationStart()``` to the new ```MelonMod.OnInitializeMelon()```.

Teo45dore committed:
Added IL2CPP/Mono cross-compatible OwOs

Teo45dore committed:
Removed old IL2CPP/Mono specific OwOs

Teo45dore committed:
Removed old csprojs and OwO.cs

Teo45dore committed:
Added updated csproj and upgraded OwO.cs

Teo45dore committed:
Updated sln to include new csproj

Teo45dore committed:
Minor code cleanup

Teo45dore committed:
Update OwOMod.csproj

Change melon loader assembly reference architecture from AMD64 to MSIL.

DubyaDude commented:
Wow, this is awesome! Has this been tested in IL2CPP and Mono environments?

DubyaDude commented on OwO/OwO.cs:
```diff
@@ -38,6 +39,13 @@ public override void OnApplicationStart()
+        //Clear cache for more randomness and strings that are not likely to appear outside their own scene.
+        public override void OnSceneWasInitialized(int buildIndex, string sceneName)
+        {
+            Utils.alreadyAppliedOwOs.Clear();
```
Build error here, 
'Utils' does not contain a definition for 'alreadyAppliedOwOs'

DubyaDude commented:
Seems to be generally working from my quick testing, commenting the problem line out.
Please fix at your earliest convenience, and I'll pull & make a release.

Teo45dore committed:
Update Utils.cs

Added cache of already OwOed strings to prevent flickering of frequently updated text.

Teo45dore committed:
Update Utils.cs

Implemented the cache )wow i love forgetting things)

Teo45dore commented:
It has been tested in IL2CPP/Mono environments! i tested in Slime Rancher 2 (IL2CPP) and in Subnautica (Mono) !

It also looks like I forgot to git add the updated Utils.cs (oops!) it should work now!

DubyaDude commented:
Made some small changes, and went ahead and made release 8: https://github.com/DubyaDude/OwO-Mod/releases/tag/Cutie-8.0
Thanks again!
